telemetry/telemetry.go

```diff
@@ -464,12 +464,13 @@ func startGNMIServer(telemetryCfg *TelemetryConfig, cfg *gnmi.Config, serverCont
 
 		serverControlValue := <-serverControlSignal
 		log.V(1).Infof("Received signal for gnmi server to close")
-		s.Stop()
 		if serverControlValue == ServerStop {
+			s.ForceStop() // No graceful stop
 			stopSignalHandler <- true
 			log.Flush()
 			return
 		}
+		s.Stop() // Graceful stop
```

Comment:
Why not stop signalHandler and flush log?

Reply:
s.Stop() is a graceful stop which is only reserved for cert rotation. Since we are not trying to exit process, just stop the server, we will not need to return or flush logs.